packages/telegram-stamper/README.md

```diff
@@ -0,0 +1,69 @@
+# @turnkey/telegram-stamper
+
+[![npm](https://img.shields.io/npm/v/@turnkey/telegram-stamper?color=%234C48FF)](https://www.npmjs.com/package/@turnkey/telegram-stamper)
+
+This package contains functions to store a turnkey api public/private key within [Telegram Cloud Storage](https://core.telegram.org/bots/webapps#cloudstorage). This package also handles stamping a Turnkey request with that api key. It is meant to be used with [`@turnkey/http`](https://www.npmjs.com/package/@turnkey/http)
+
+Preqrequisites: \
+Telegram Bot API >= 6.9
+
+Usage:
+
+Insert a new api key into Telegram Cloud Storage
+
+```ts
+import { TelegramStamper } from "@turnkey/telegram-stamper";
+import { TurnkeyClient } from "@turnkey/http";
+
+// create a new telegram stamper
+const stamper = new TelegramStamper({
+  apiPublicKey: "...",
+  apiPrivateKey: "...",
+});
+
+// initialize the telegram stamper, this stores the api key credentials in Telegram Cloud Storage for the first time
+try {
+  stamper.init();
+} catch (err) {
+  throw new Error(`Failed initializing Telegram Stamper: ${err}`)
+}
```

Comment:
Having said that, maybe its not super likely thats how the tgram stamper will be used? Maybe this should just be removed overall? (from the readme)

Reply:
yep yep, I get that. But it could create an API key if nothing's in cloudStorage right? From a dev's POV the construction becomes the same regardless of whether a user has a key or not.

Currently devs will be forced to branch:
```ts
if (hasApiKeyInCloudStorage) {
   const stamper = new TelegramStamper({});
} else {
   const stamper = new TelegramStamper({
     apiPublicKey: "...", // also need to worry about how to generate it
     apiPrivateKey: "...",
   }
}
```

Whereas if we had this behavior (generate key when storage is empty), the init simplifies to:
```ts
const stamper = new TelegramStamper({});
```

I guess this is only important if creating and persisting a **new** key is a common pattern. If the stamper is almost always initiated with an existing key from somewhere else, not worth worrying about.

Reply:
To be clear, the tgram stamper does not create any api keys for users, they need to be passed to the stamper

Reply:
OOp I wrote that comment before I reloaded and saw your response, but yeah Im not exactly sure it makes sense for the stamper to actually create the api keys, because those api keys will have to be added to the users account somehow meaning that the tgram stamper already needs a valid api key, or something needs to pass the tgram stamper a valid key

Reply:
Cool! I'm okay with that. Let's not bundle API key creation in there and always expect public/private key. As you build out the TG mini-app I think patterns will naturally emerge anyway. I suspect you'll need to create a stamper from a bundle (coming from oauth or email auth). That might be a convenience function we want to provide (`TelegramStamper.fromBundle(...)` would decrypt the bundle and store it in cloudstorage in one go)

Reply:
Yeah, I think thatll be a common pattern for sure, that sounds like a nice convenience for sure!

Reply:
by the way this brings up a tangential point: it's probably best not to merge this PR until the TG mini-app is built: that way you can get a "feel" for the interface you're building with this stamper, hammer out the bug, etc. The iteration loop I'd recommend is:
1. TG stamper PR opened (what you have here)
1. TG mini app repo or branch somewhere which imports the codesandbox version of this package
1. Try your mini app in TG, look for bugs; look at the code of the mini-app, look for awkwardness. Go back to step 1 and edit to solve for bugs and awkwardness! 

and when we're convinced this stamper is a fit for the mini-app: merge PR, publish to NPM, and import the final package in the mini app.

Reply:
🫡